File: tests/fixtures/sample-js-project-il/code/src/events/bus.ts
import { EventEmitter } from 'node:events';

// ADR-003 mandates Kafka for inter-service messaging, but the event bus is
// an in-process Node EventEmitter and no Kafka client ships at all.
// IL-DRIFT: ArchitectureDecision:messaging.kafka / architecture.messaging.unmet-choice
export const events = new EventEmitter();

export interface OrderEvent {
  id: string;
  status: string;
  at: string;
}

File: tests/fixtures/sample-python-project-il/code/app/events/bus.py
# ADR-003 mandates Kafka for inter-service messaging, but the event bus is
# an in-process callback registry and no Kafka client ships at all.
# IL-DRIFT: ArchitectureDecision:messaging.kafka / architecture.messaging.unmet-choice
_subscribers: dict = {}


def emit(name: str, payload: dict) -> None:
    for cb in _subscribers.get(name, []):
        cb(payload)
